src/main/java/sg/ncl/MainController.java

```diff
@@ -839,11 +828,11 @@ public String validateDetails(
                                 InvalidPasswordException |
                                 DeterLabOperationFailedException e) {
                     redirectAttributes.addFlashAttribute(MESSAGE, e.getMessage());
-                    redirectAttributes.addFlashAttribute("signUpMergedForm", signUpMergedForm);
+                    redirectAttributes.addFlashAttribute(SIGNUP_MERGED_FORM, signUpMergedForm);
                     return "redirect:/signup2";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "redirect:/signup2" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)